src/context.ml

```diff
@@ -540,11 +540,42 @@ let create ~(kind : Kind.t) ~path ~env ~env_nodes ~name ~merlin ~targets
   in
   native :: List.filter_opt others
 
+let extend_paths t ~env =
+  let module Eval =
+    Ordered_set_lang.Make(String)
+      (struct
+        type t = string
+        type key = string
+        let key x = x
+      end)
+  in
+  let t =
+    let f (var, t) =
+      let parse ~loc:_ s = s in
+      let standard = Env.path env |> List.map ~f:Path.to_string in
+      var, Eval.eval t ~parse ~standard
+    in
+    List.map ~f t
+  in
+  let vars =
+    let cwd = Sys.getcwd () in
+    let of_filename_relative_to_cwd s =
+      if Filename.is_relative s then
+        Filename.concat cwd s
+      else
+        s
+    in
+    let sep = String.make 1 Bin.path_sep in
+    let f l = String.concat ~sep (List.map ~f:of_filename_relative_to_cwd l) in
+    Env.Map.of_list_reduce t ~f:List.append |> Env.Map.map ~f
```

Comment:
Does this `of_list_reduce` means that we can have the same variable several times inside `(paths ...)`?

i.e.:

```
(paths
 (PATH ...)
 (PATH ...)
)
```
?

That feels dodgy. We should forbid such cases.

Reply:
Yes, it means that; and yes, it is dodgy. Will amend the code to fail in that case.

Reply:
This is now fixed.